extension/spark-sink/src/main/scala/org/apache/spark/sql/palo/PaloSinkProvider.scala

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.palo
+
+import org.apache.spark.internal.Logging
+import org.apache.spark.sql.{AnalysisException, SQLContext}
+import org.apache.spark.sql.execution.streaming.Sink
+import org.apache.spark.sql.sources.{DataSourceRegister, StreamSinkProvider}
+import org.apache.spark.sql.streaming.OutputMode
+
+/**
+ * The provider class for the [[PaloSink]].
+ * Check the necessary parameters as soon as the job start on driver.
+ * If the parameters miss, will throw IllegalArgumentException.
+ */
+private[palo] class PaloSinkProvider extends StreamSinkProvider
+    with DataSourceRegister with Logging {
```

Comment:
separate DataSourceRegister with another class maybe better.
such as class DefaultSource15 extends DefaultSource with DataSourceRegister

DefaultSource is implemention